TBCCore/TBCKVOObservation.m

```diff
@@ -0,0 +1,55 @@
+//  Copyright © 2018 Tabcorp Pty. Ltd. All rights reserved.
+
+#import "TBCKVOObservation.h"
+
+@implementation TBCKVOObservation {
+    __weak NSObject * _object;
+    NSString * _keyPath;
+    TBCKVOObservationBlock _block;
+    BOOL _observing;
+}
+
++ (instancetype)observationWithSelector:(SEL)selector object:(NSObject *)object block:(TBCKVOObservationBlock)block {
+    return [[self alloc] initWithSelector:selector object:object block:block];
+}
+
+- (instancetype)initWithSelector:(SEL)selector object:(NSObject * __nullable)object options:(NSKeyValueObservingOptions)options block:(TBCKVOObservationBlock)block {
```

Comment:
Why `selector` over `keyPath`?

Reply:
I'd've put object first too, but whatevs.

Reply:
Internally, we've adopted using NSStringFromSelector(@selector(...)) when KVOing, giving the compiler the chance to complain when you make an error, or refactor.
Given that this is generic perhaps it's best to provide both.